link.go

```diff
@@ -1,12 +1,31 @@
 package provider
 
+// NOTE(brooksmtownsend): There might be a better way to represent this in Go, please comment
+// or leave an issue if you can think of one. Perhaps I could do the decryption during the
+// unmarshalling process, but I'm not sure if that would be a good idea.
```

Comment:
This is one of the less ideal parts of this PR, so I'd absolutely love some advice here. The need for this struct is to unmarshal all the link related information and the encrypted secret bytes, then decrypt those bytes and finally unmarshal into the final InterfaceLinkDefinition structure.

Reply:
Yeah because you need the key I don't think there is a way around this because you need the key to decrypt

Reply:
For any future onlookers, improvement here welcome.